lib/resources/package.rb

```diff
@@ -71,13 +73,28 @@ def version
     def to_s
       "System Package #{@package_name}"
     end
+
+    private
+
+    def evaluate_missing_requirements(missing_requirements)
```

Comment:
You want to bail out of a method as quickly as possible if you can. In this case, you build up a `skip_msg` that you may never use because you don't check for emptiness of the missing requirements until the very last step.

I'd rewrite this method like this:

```ruby
def evaluate_missing_requirements
  missing_requirements = @pkgman.missing_requirements
  return if missing_requirements.empty?

  skip_resource "The following requirements are not met for this resource: #{missing_requirements.join(', ')}"
end
```

I don't know that you need to compact and uniq unless you're doing something in your code that you think would require that... in which case I'd encourage you to solve that problem since you control the whole kit and kaboodle. 🙂